contracts/exchangeIssuance/ExchangeIssuanceLeveragedForCompound.sol

```diff
@@ -0,0 +1,1346 @@
+        );
+
+        _withdrawFromCompound(
+            _decodedParams.leveragedTokenData.collateralToken,
+            _decodedParams.leveragedTokenData.collateralAmount
+        );
+        // Obtain debt tokens required to repay flashloan by swapping the underlying collateral tokens obtained in withdraw step
+        uint256 collateralTokenSpent = _swapCollateralForDebtToken(
+            _debtTokenAmountNet + _premium,
+            _debtToken,
+            _decodedParams.leveragedTokenData.collateralAmount,
+            _decodedParams.leveragedTokenData.collateralToken,
+            _decodedParams.collateralAndDebtSwapData
+        );
+        // Liquidate remaining collateral tokens for the payment token specified by user
+        uint256 amountOutputToken = _liquidateCollateralTokens(
+            collateralTokenSpent,
+            _decodedParams.setToken,
+            _decodedParams.setAmount,
+            _decodedParams.originalSender,
+            _decodedParams.paymentToken,
+            _decodedParams.limitAmount,
+            _decodedParams.leveragedTokenData.collateralToken,
+            _decodedParams.leveragedTokenData.collateralAmount  - 2*ROUNDING_ERROR_MARGIN,
+            _decodedParams.paymentTokenSwapData
+        );
+        require(amountOutputToken >= _decodedParams.limitAmount, "ExchangeIssuance: INSUFFICIENT OUTPUT AMOUNT");
+    }
+
+
+    /**
+    * Returns the collateral / debt token addresses and amounts for a leveraged index
+    *
+    * @param _setToken              Address of the SetToken to be issued / redeemed
+    * @param _setAmount             Amount of SetTokens to issue / redeem
+    * @param _isIssuance            Boolean indicating if the SetToken is to be issued or redeemed
+    *
+    * @return Struct containing the collateral / debt token addresses and amounts
+    */
+    function _getLeveragedTokenData(
+        ISetToken _setToken,
+        uint256 _setAmount,
+        bool _isIssuance
+    )
+    internal
+    view
+    returns (LeveragedTokenData memory)
+    {
+        address[] memory components;
+        uint256[] memory equityPositions;
+        uint256[] memory debtPositions;
+        address _collateralToken;
+        LeveragedTokenData memory _leveragedTokenData;
+        uint256 requiredCTokenAmount;
+
+        if(_isIssuance){
+            (components, equityPositions, debtPositions) = debtIssuanceModule.getRequiredComponentIssuanceUnits(_setToken, _setAmount);
+        } else {
+            (components, equityPositions, debtPositions) = debtIssuanceModule.getRequiredComponentRedemptionUnits(_setToken, _setAmount);
+        }
+
+        
+        require(equityPositions[0] == 0 || equityPositions[1] == 0, "ExchangeIssuance: TOO MANY EQUITY POSITIONS");
+        
+        require(debtPositions[0] == 0 || debtPositions[1] == 0, "ExchangeIssuance: TOO MANY DEBT POSITIONS");
+
+        if(equityPositions[0] > 0){
+            if (components[0] == cEtherADDRESS) {
```

Comment:
This (the whole outer if / else block) looks like a lot of code duplication to me. The difference in the two if/else branches seems only to be `components[0]` vs `components[1]` etc.  I think you could simplify this code a if you assigned the collateral  / debt token in the beginning. Might even save some gas since you'd be doing a lot fewer array accesses.

Reply:
addressed in 2bc43dfd123d9c826c22e1931048af595bb3ebc3